src/components/settings/sections/models/ChatModelsSubSection.tsx

```diff
@@ -31,18 +34,35 @@ export function ChatModelsSubSection({
     })
   }
 
+  const handleToggleEnableChatModel = async (modelId: string) => {
+    await setSettings({
+      ...settings,
+      chatModels: [...settings.chatModels].map((v) =>
+        v.id === modelId ? { ...v, enable: !isEnabled(v.enable) } : v,
+      ),
+    })
+  }
```

Comment:
If you use ObsidianToggle component, there’s no need to define it using negation. Simply using the value makes it clearer.

Reply:
Additionally, the function should handle the case where the user attempts to disable a model that is currently selected as the Chat Model or Apply Model. It should display a notice and do nothing, similar to `handleDeleteChatModel`.

Reply:
> Additionally, the function should handle the case where the user attempts to disable a model that is currently selected as the Chat Model or Apply Model. It should display a notice and do nothing, similar to `handleDeleteChatModel`.

Thank you for catching this edge case! I've added a validation check to prevent disabling a model that is currently selected as either the Chat Model or Apply Model. The function now shows a notice to the user in such cases, following the same pattern as handleDeleteChatModel.